### PR TITLE
Add OPTIMADE querier

### PR DESCRIPTION
**Pull request by ml-evs**
Just a draft PR that I have not tested yet. You may want to try running the OPTIMADE query if you already have the package installed, otherwise I'll put in some work trying to get it installed at my end.

I guess running the tests with continuous integration, e.g. GitHub actions, would be useful?

### CODE REVIEW

**Comment by pedrobcst** [comment]
I will check it soon. Sorry, I have few personal things I have to handle so I hope to check and finish this implementation soon. Regarding CI, Ive looked around it but I could not find a way to set it up. Specially how to automatically start mongo, create a conda env, do pip, and set up all the config files automatically. If you have any clues on how to do it let me know. I was also planning to create a docker enviroment to easily deploy it but I got stuck on the same issue.

**Comment by ml-evs** [comment]
> I will check it soon. Sorry, I have few personal things I have to handle so I hope to check and finish this implementation soon. Regarding CI, Ive looked around it but I could not find a way to set it up. Specially how to automatically start mongo, create a conda env, do pip, and set up all the config files automatically. If you have any clues on how to do it let me know. I was also planning to create a docker enviroment to easily deploy it but I got stuck on the same issue.

This is something I have quite a lot of experience with (it is fiddly). You can see an example here: https://github.com/Materials-Consortia/optimade-python-tools/blob/master/.github/workflows/ci.yml, where we test the optimade-python-tools package with a series of database backends. Here is how to set up mongo, for e.g., https://github.com/Materials-Consortia/optimade-python-tools/blob/bb3c5a11364cce5c8e8d4c2019c9673a0e77d9ea/.github/workflows/ci.yml#L147-L151.

I'll raise another issue about this and will see what time I can find.

**Comment by pedrobcst** [comment]
I might have pressed the wrong button, but I will start working on this implementation, thanks for the draft. I will put some comments in the Issues section though.

**Comment by pedrobcst** [comment]
It seems I have mistakenly created a new branch, however my implementation is at

https://github.com/pedrobcst/Xerus/blob/ml-evs-ml-evs/add_optimade/Xerus/queriers/optimade.py,

There are still issues to be solved, as I listed in the issues section.

**Comment by ml-evs** [comment]
> 

I think the confusion is that I committed from a fork, I can give you access to it and you should be able to push your changes to my branch with something like


```
git remote add upstream git@github.com:ml-evs/Xerus
git checkout ml-evs/ml-evs/add_optimade   # (your branch)
git push upstream -u ml-evs/add_optimade
```

and the changes will appear here. Alternatively we can just merge this PR and start a new one based on your additions (though I will then not have access to add new changes, unless you add me as a collaborator for the repo).

**Comment by pedrobcst** [comment]
I see. Sorry for the confusion. Either way is fine for me. What I will do now is to implement your changes, test it a bit, and I will push it to your fork them. I have no problem in adding you as a collabotor here though if it is better (it actually might be better for me).
Just give me some time to fix few things

**Comment by ml-evs** [comment]
Cool, I just added you as a collaborator on the fork, but feel free to do as you see fit

File: Xerus/queriers/optimade.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        os.mkdir(self.folder_path, exist_ok=True)
+
+    def make_name(self, entry, meta) -> str:
+        return f'{meta["provider"]["prefix"]}_{entry["id"]}.cif'
+
+
+    def query(self, query_url: str | None) -> None:
+
+        if not query_url:
+            query_url = f"{self.base_url}/{self.optimade_endpoint}?{self.optimade_filter}&{self.optimade_response_fields}"
```
Comment (ml-evs) [inline Xerus/queriers/optimade.py:49]:
```suggestion
            query_url = f"{self.base_url}/{self.optimade_endpoint}?{self.optimade_filter}&{self.optimade_response_fields}&page_limit=10"
```

**Comment by pedrobcst** [comment]
Hey Evan,

I think it will be much easier with you work directly here. I will add you as colaborator and he work directly here. Ive merged my branch now with most changes we did so I will close this PR and we can work directly here

**Comment by ml-evs** [comment]
Can we rebase this PR and merge it, instead of closing? I think one commit is missing from the one that was merged (re: page_limit)

**Comment by ml-evs** [comment]
> Can we rebase this PR and merge it, instead of closing? I think one commit is missing from the one that was merged (re: page_limit)

Nevermind, I see you added it yourself!

**Pull request closed without merging**